rust/storage/src/s3.rs

```diff
@@ -640,21 +639,8 @@ mod tests {
             .await
             .unwrap();
 
-        let mut stream = storage.get("test").await.unwrap();
-
-        let mut buf = Vec::new();
-        while let Some(chunk) = stream.next().await {
-            match chunk {
-                Ok(data) => {
-                    buf.extend_from_slice(&data);
-                }
-                Err(e) => {
-                    panic!("Error reading stream: {}", e);
-                }
-            }
-        }
-
-        let buf = String::from_utf8(buf).unwrap();
+        let buf = storage.get("test").await.unwrap();
+        let buf = String::from_utf8_lossy(&buf);
```

Comment:
nit: does this have to be `from lossy`? could be confusing to a reader assuming thats deliberate

Reply:
`from_utf8` takes ownership, can update